tools/checkpoint_conversion/convert_xlnet_checkpoints.py

```diff
@@ -0,0 +1,303 @@
+# Copyright 2023 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import copy
+import os
+
+import h5py
+import numpy as np
+import requests
+import tensorflow as tf
+from transformers import TFXLNetModel
+from transformers import XLNetTokenizer
+
+from keras_nlp.models import XLNetBackbone
+
+check_mems = False
+
+PRESET = "xlnet-base-cased"
+CKPT = f"https://huggingface.co/{PRESET}"
+SAVE_PATH = "./tf_weights.h5"
+
+# create HF model
+hf_model = TFXLNetModel.from_pretrained(PRESET)
+
+print(f"GPU Available or not : {tf.test.is_gpu_available()}")
+
+with open(SAVE_PATH, "wb") as p:
+    response = requests.get(CKPT + "/resolve/main/tf_model.h5")
+    p.write(response.content)
+
+
+tokenizer = XLNetTokenizer.from_pretrained(PRESET)
+string = "An input text string."
+tokens = tokenizer(string, return_tensors="tf", return_attention_mask=True)
+# tokens["target_mapping"] = tf.random.uniform(
```

Comment:
Why all the commented out code?

Reply:
The commented out code are the optional inputs, since we don't use optional inputs any more for the backbone I will remove them in the next commit.